### ISSUE TITLE
week groupings are indicating wrong year on first week of year

### ISSUE THREAD

**Issue by robinfrick** [metabase/metabase#1987]
When grouping by week the time around the year change throws the query builder off track. The last days of 2015 and first days of 2016 are labeled as "1st - 2015". Apart from that being wrong it continues counting with "2nd - 2016". This leads to weeks being off by one. I can already see values that are supposedly from the 9th week of 2016 although the 8th week has just started.
![image](https://cloud.githubusercontent.com/assets/8144392/13218706/97f6417c-d96b-11e5-8aba-bf75e731d1c2.png)
![image](https://cloud.githubusercontent.com/assets/8144392/13218707/9aacc882-d96b-11e5-9812-f9ab08e099fe.png)

A solution would be to switch to iso weeks (https://en.wikipedia.org/wiki/ISO_week_date#First_week). Apart from being an iso standard this has the additional advantage for me as someone who is from "anywhere but the US" that weeks start on Monday like they should ;-)

**Comment by camsaul** [comment]
The first week of the year can end up starting the year before, e.g. if the first day of the week is Sunday and the year starts on a Tuesday then the first week of the year is considered to have started that Sunday the prior year. So the display issue probably has something to do with that.

**Comment by tlrobinson** [comment]
moment.js has a "ISO 4 digit week year" which should be what we want. http://momentjs.com/docs/#/displaying/

**Comment by tlrobinson** [comment]
Do we want to use "local week of year" or "ISO week of year"? Either way as long as we use the corresponding year token the year will match up with the week (thus resolving this issue)

**Comment by camsaul** [comment]
FWIW Sunday is the first day of the week in a lot of places besides the US. Currently the only week grouping we support is Sunday as the first day of the week, mostly because that's standard used by most database databases and in Java.
![screen shot 2016-02-22 at 12 27 32 pm](https://cloud.githubusercontent.com/assets/1455846/13231644/b6351fc0-d95f-11e5-9eaf-fee9969b7a23.png)

The QP always returns weeks of years starting on Sunday. So hopefully there's a week of year function that will let you match up to the weeks starting on Sunday. Otherwise "ISO week" will be incorrect everywhere and "local week of year" will be incorrect in about half of the world.

**Comment by camsaul** [comment]
Grouping by weeks with a non-Sunday starting day is already opened as a separate issue, #1779

**Comment by tlrobinson** [comment]
It looks like in moment.js it's possible to use a specific locale when parsing/formatting, but not to pick a specific week start day. For now I think we can just use the US locale. When we fix #1779 we'll have to figure out something else.

**Comment by robinfrick** [comment]
Sorry @camsaul I thought my unrepresentative sample of countries I lived in would serve me better. According to your graphic I just had the luck choosing the Monday countries. Especially for Paraguay, the only country I lived in in the Americas was a very lucky choice.

But yes I agree the whole Monday/Sunday-start of the week thing isn't as important as having the weeks numbered correctly and not have the first week as the last week of a year. It's either the 53rd week of 2015 or the 1st week of 2016, but 1st week of 2015 doesn't make a lot of sense in this case.
